tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1638,6 +1643,96 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 			}
 		})
 	})
+
+	ginkgo.Describe("[Feature]: TopologySpreadConstraint", func() {
+		nodeZoneMap := map[string]string{}
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("add or update labels to node workers")
+
+			nodeList, err := c.CoreV1().Nodes().List(metav1.ListOptions{})
+			framework.ExpectNoError(err, "failed to list nodes: %+v")
+			for i := range nodeList.Items {
+				node := &nodeList.Items[i]
+				zone := "zone-" + strconv.Itoa(i%2)
+				patch := []byte(fmt.Sprintf(
+					`{"metadata":{"labels":{"%s":"%s"}}}`,
+					LabelKeyTestingZone,
+					zone,
+				))
+				_, err = c.CoreV1().Nodes().Patch(node.Name, types.StrategicMergePatchType, patch)
```

Comment:
Can we only set labels once for the whole E2E test? if we add more `It`s with node modify and they run concurrently in multi Ginkgo nodes, the labels may be modified by other `BeforeEach`.

Reply:
Fixed